optaplanner-core/src/main/java/org/optaplanner/core/config/constructionheuristic/placer/QueuedEntityPlacerConfig.java

```diff
@@ -40,22 +36,16 @@
 import org.optaplanner.core.config.heuristic.selector.move.generic.chained.SubChainChangeMoveSelectorConfig;
 import org.optaplanner.core.config.heuristic.selector.move.generic.chained.SubChainSwapMoveSelectorConfig;
 import org.optaplanner.core.config.heuristic.selector.move.generic.chained.TailChainSwapMoveSelectorConfig;
-import org.optaplanner.core.config.heuristic.selector.value.ValueSelectorConfig;
 import org.optaplanner.core.config.util.ConfigUtils;
-import org.optaplanner.core.impl.constructionheuristic.placer.QueuedEntityPlacer;
-import org.optaplanner.core.impl.domain.entity.descriptor.EntityDescriptor;
-import org.optaplanner.core.impl.domain.variable.descriptor.GenuineVariableDescriptor;
+import org.optaplanner.core.impl.constructionheuristic.placer.QueuedEntityPlacerFactory;
 import org.optaplanner.core.impl.heuristic.HeuristicConfigPolicy;
-import org.optaplanner.core.impl.heuristic.selector.entity.EntitySelector;
-import org.optaplanner.core.impl.heuristic.selector.move.MoveSelector;
-import org.optaplanner.core.impl.heuristic.selector.move.MoveSelectorFactory;
 
 public class QueuedEntityPlacerConfig extends EntityPlacerConfig<QueuedEntityPlacerConfig> {
 
     public static QueuedEntityPlacerConfig unfoldNew(HeuristicConfigPolicy configPolicy,
             List<MoveSelectorConfig> templateMoveSelectorConfigList) {
         QueuedEntityPlacerConfig config = new QueuedEntityPlacerConfig();
-        config.entitySelectorConfig = config.buildEntitySelectorConfig(configPolicy);
+        config.entitySelectorConfig = new QueuedEntityPlacerFactory(config).buildEntitySelectorConfig(configPolicy);
```

Comment:
The fact that the unfold methods create a new PlacerFactory,
as well as do a lot of defaulting (configuration by exception),
makes me think we might want to move them to the Factory side too, alongside the build methods?

Reply:
The "configuration by exception" in one place argument is really the big reason I'd like to seem them in the build methods.

Due to conf by exception, most properties are automatically assigned a default value (sometimes calculated to a default value). For example, a move selector doesn't have to explicity state its cacheType in the XML (it can, but it doesn't have too).
Often, we need to answer "what does that property default too again?" That used to have 1 answer: in the Config class. Now, if the unfold methods are in the Config class, it depends. I'd rather see all that information in 1 location, the new Factory class.